src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -0,0 +1,60 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect} from 'react';
+import {Linking} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx/lib/types';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import * as SessionUtils from '@libs/SessionUtils';
+import type {AuthScreensParamList} from '@navigation/types';
+import * as SessionActions from '@userActions/Session';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Session} from '@src/types/onyx';
+
+type LogOutPreviousUserPageOnyxProps = {
+    /** The data about the current session which will be set once the user is authenticated and we return to this component as an AuthScreen */
+    session: OnyxEntry<Session>;
+};
+
+type LogOutPreviousUserPageProps = LogOutPreviousUserPageOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
+
+// This page is responsible for handling transitions from OldDot. Specifically, it logs the current user
+// out if the transition is for another user.
+//
+// This component should not do any other navigation as that handled in App.setUpPoliciesAndNavigate
+function LogOutPreviousUserPage({session, route}: LogOutPreviousUserPageProps) {
+    useEffect(() => {
+        Linking.getInitialURL().then((transitionURL) => {
+            const sessionEmail = session?.email;
+            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL ?? undefined, sessionEmail);
+
+            if (isLoggingInAsNewUser) {
+                SessionActions.signOutAndRedirectToSignIn();
+            }
+
+            // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
+            // and their authToken stored in Onyx becomes invalid.
+            // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
+            // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
+            const shouldForceLogin = route.params.shouldForceLogin === 'true';
+            if (shouldForceLogin) {
+                const email = route.params.email;
+                const shortLivedAuthToken = route.params.shortLivedAuthToken;
```

Comment:
I think we should default these to empty strings, since the action function expects strings.

Reply:
done